### PR TITLE
chore(semver): replace `replace()` with `match()` capturing groups 

### PR DESCRIPTION
**Pull request by timreichen**
Replaces `replace()` functions with `match()` and uses groups for more readability.

### CODE REVIEW

File: semver/parse_range.ts [hunk 1]
```diff
@@ -64,44 +63,52 @@ function replaceCarets(comp: string): string {
...
+        return `>=${groups.major}.${groups.minor}.${groups.patch}-${groups.prerelease} <${groups.major}.${groups.minor}.${
+          +groups.patch + 1
+        }`;
       } else {
-        ret = `>=${M}.${m}.${p} <${+M + 1}.0.0`;
+        return `>=${groups.major}.${groups.minor}.${groups.patch}-${groups.prerelease} <${groups.major}.${
+          +groups.minor + 1
+        }.0`;
       }
+    } else {
```
Comment (iuioiua) [inline semver/parse_range.ts:92]: It'd be nice to cleanup these `else` blocks in another PR.

**Review by iuioiua — APPROVED** [review]
Nice improvement in readability. LGTM.

File: semver/parse_range.ts [hunk 2]
```diff
@@ -25,28 +25,27 @@ function replaceTildes(comp: string): string {
...
+    return "";
+  } else if (isWildcard(groups.minor)) {
+    return `>=${groups.major}.0.0 <${+groups.major + 1}.0.0`;
+  } else if (isWildcard(groups.patch)) {
+    // ~1.2 == >=1.2.0 <1.3.0
+    return `>=${groups.major}.${groups.minor}.0 <${groups.major}.${
+      +groups.minor + 1
+    }.0`;
+  } else if (groups.prerelease) {
+    return `>=${groups.major}.${groups.minor}.${groups.patch}-${groups.prerelease} <${groups.major}.${
```
Comment (kt3k) [inline semver/parse_range.ts:41]: These lines look a bit long. How about destructuring `major` `minor` `patch` `prerelease` at line 30?

File: semver/parse_range.ts (hunk 1 above)
Comment (timreichen) [reply semver/parse_range.ts:87]: I'll do that in another PR

**Review by iuioiua — APPROVED** [review]

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**